components/ipfs/ipfs_component_cleaner.cc

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/components/ipfs/ipfs_component_cleaner.h"
+
+#include "base/files/file_util.h"
+#include "base/path_service.h"
+#include "base/task/thread_pool.h"
+#include "build/build_config.h"
+#include "chrome/common/chrome_paths.h"
+
+namespace {
+#if BUILDFLAG(IS_WIN)
+static const base::FilePath::StringPieceType kIpfsClientComponentId =
+    FILE_PATH_LITERAL("lnbclahgobmjphilkalbhebakmblnbij");
+#elif BUILDFLAG(IS_MAC)
+#if defined(ARCH_CPU_ARM64)
+static const base::FilePath::StringPieceType kIpfsClientComponentId =
+    FILE_PATH_LITERAL("lejaflgbgglfaomemffoaappaihfligf");
+#else
+static const base::FilePath::StringPieceType kIpfsClientComponentId =
+    FILE_PATH_LITERAL("nljcddpbnaianmglkpkneakjaapinabi");
+#endif
+#elif BUILDFLAG(IS_LINUX)
+#if defined(ARCH_CPU_ARM64)
+static const base::FilePath::StringPieceType kIpfsClientComponentId =
+    FILE_PATH_LITERAL("fmmldihckdnognaabhligdpckkeancng");
+#else
+static const base::FilePath::StringPieceType kIpfsClientComponentId =
+    FILE_PATH_LITERAL("oecghfpdmkjlhnfpmmjegjacfimiafjp");
+#endif
+#else
+// Not used yet for Android/iOS
+static const base::FilePath::StringPieceType kIpfsClientComponentId =
+    FILE_PATH_LITERAL("");
```

Comment:
I think it is better to delete this, otherwise looks like whole user data dir could be deleted on this platforms.

Reply:
Or add static assert